internal/bminventory/inventory.go

```diff
@@ -1710,9 +1710,15 @@ func (b *bareMetalInventory) updateClusterData(ctx context.Context, cluster *com
 			updates["pull_secret_set"] = false
 		}
 	}
-	if params.ClusterUpdateParams.APIVipDNSName != nil && swag.StringValue(cluster.Kind) == models.ClusterKindAddHostsCluster {
-		log.Infof("Updating api vip to %s for day2 cluster %s", *params.ClusterUpdateParams.APIVipDNSName, cluster.ID)
-		updates["api_vip_dns_name"] = *params.ClusterUpdateParams.APIVipDNSName
+
+	if params.ClusterUpdateParams.APIVipDNSName != nil {
+		if swag.StringValue(cluster.Kind) == models.ClusterKindAddHostsCluster {
+			log.Infof("Updating api vip to %s for day2 cluster %s", *params.ClusterUpdateParams.APIVipDNSName, cluster.ID)
+			updates["api_vip_dns_name"] = *params.ClusterUpdateParams.APIVipDNSName
+		} else {
+			log.Errorf("Can't update api vip to %s for day1 cluster %s", *params.ClusterUpdateParams.APIVipDNSName, cluster.ID)
+			return common.NewApiError(http.StatusBadRequest, err)
```

Comment:
what is `err`? it should  be the same as the log.
You can define 
```suggestion
			msg := "Can't set API VIP DNS name"
			log.Error(msg)
			return common.NewApiError(http.StatusBadRequest, errors.Errorf(msg))
```